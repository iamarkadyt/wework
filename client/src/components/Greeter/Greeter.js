import React from 'react'
import {
    FaShippingFast as IcoTruck
} from 'react-icons/fa'
import './Greeter.css'

class Greeter extends React.Component {
    state = {
        startAnimations: false
    }

    handleLoad = () => {
        this.setState({ startAnimations: true })
    }

    render() {
        const { startAnimations } = this.state

        const whiteScreenClasses = [
            "Greeter__white-screen",
            startAnimations && "animate"
        ].join(' ')

        const logoScreenClasses = [
            "Greeter__logo-screen",
            startAnimations && "animate"
        ].join(' ')

        const headlineClasses = [
            "Greeter__headline",
            startAnimations && "animate"
        ].join(' ')

        const headlineWarningClasses = [
            "Greeter__headline_warning",
            startAnimations && "animate"
        ].join(' ')

        return (
            <div className="Greeter__container">
                <div className={logoScreenClasses}>
                    <div className="Greeter__logo">
                        <span>We</span><span>Work</span>
                    </div>
                    <span className={headlineClasses}>
                        Social network website for professionals
                    </span>
                    <span className={headlineWarningClasses}>
                        Support for the tall screens is already<br />
                        on the way!<br /><br />

                        <span style={{fontSize: 80}}><IcoTruck /></span><br /><br />

                        Until then you can visit the website<br />
                        on desktop or switch your<br />
                        device into the landscape mode.<br /><br />

                        To checkout my skills of building the<br />
                        responsive layouts visit&nbsp;
                        <a href="https://arkadyt.com" target="_blank" rel="noopener noreferrer">
                            <b>https://arkadyt.com</b>
                        </a>!<br />
                    </span>
                </div>
                <div className={whiteScreenClasses} />
            </div>
        )
    }

    componentDidMount() {
        window.addEventListener('load', this.handleLoad)
        document.documentElement.className = 'html-dark-background'
    }
}

export default Greeter