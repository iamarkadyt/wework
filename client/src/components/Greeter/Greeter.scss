@import '../../_variables.scss';
@import '../../_mixins.scss';
.Greeter__container {
    position: relative;
    z-index: 205;
    .Greeter__white-screen {
        @include viewportSheet();
        background-color: white;
        &.animate {
            animation: fadeOut .5s ease-out 0s 1 normal forwards;
        }
    }
    .Greeter__logo-screen {
        @include viewportSheet();
        background-color: $color-bg-main;
        &.animate {
            @media (orientation: landscape) {                
                animation: fadeOut .5s ease-out 3.75s 1 normal forwards;
            }
        }
        display: flex;
        flex-direction: column;
        justify-content: center;
        align-items: center;
        .Greeter__logo {
            font-size: 5rem;
            font-weight: bold;
            color: $color-font-main;
            span:last-child {
                color: $color-accent-main;
            }
        }
        .Greeter__headline,
        .Greeter__headline_warning {
            color: $color-font-main;
            font-size: 1.5rem;
            opacity: 0;
            text-align: center;
            &.animate {
                animation: fadeOut 1s ease-out 1s 1 reverse forwards;
            }
        }
        .Greeter__headline_warning {
            @media (orientation: landscape) {
                display: none;
            }
            a {
                color: white;                
            }
        }
        .Greeter__headline {
            @media (orientation: portrait) {
                display: none;
            }
        }
    }
}

@keyframes fadeOut {
    from {
        opacity: 1;
    }
    to {
        opacity: 0;
        pointer-events: none;
    }
}
