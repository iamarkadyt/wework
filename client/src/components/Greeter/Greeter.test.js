import React from 'react'
import { shallow } from 'enzyme'
import Greeter from './Greeter'
import cloneDeep from 'lodash.clonedeep'

let mountedComponent, props

const getMockProps = () => {
  // left here in case of future changes to the component
  return cloneDeep({})
}

const comp = () => {
  if (!mountedComponent) {
    mountedComponent = shallow(<Greeter {...props} />)
  }
  return mountedComponent
}

describe('Greeter', () => {
  beforeEach(() => {
    mountedComponent = undefined
    props = getMockProps()
  })

  describe('state', () => {
    describe('startAnimations', () => {
      it('if true, whiteScreenClasses gets additional className', () => {
        comp().setState({ startAnimations: true })
        expect(comp().find('.Greeter__white-screen').hasClass('animate')).toBe(true)
      })

      it('if false, whiteScreenClasses does not get additional classNames', () => {
        comp().setState({ startAnimations: false })
        expect(comp().find('.Greeter__white-screen').hasClass('animate')).toBe(false)
      })

      it('if true, logoScreenClasses gets additional className', () => {
        comp().setState({ startAnimations: true })
        expect(comp().find('.Greeter__logo-screen').hasClass('animate')).toBe(true)
      })

      it('if false, logoScreenClasses does not get additional classNames', () => {
        comp().setState({ startAnimations: false })
        expect(comp().find('.Greeter__logo-screen').hasClass('animate')).toBe(false)
      })

      it('if true, headlineClasses gets additional className', () => {
        comp().setState({ startAnimations: true })
        expect(comp().find('.Greeter__headline').hasClass('animate')).toBe(true)
      })

      it('if false, headlineClasses does not get additional classNames', () => {
        comp().setState({ startAnimations: false })
        expect(comp().find('.Greeter__headline').hasClass('animate')).toBe(false)
      })

      it('if true, headlineWarningClasses gets additional className', () => {
        comp().setState({ startAnimations: true })
        expect(comp().find('.Greeter__headline_warning').hasClass('animate')).toBe(true)
      })

      it('if false, headlineWarningClasses does not get additional classNames', () => {
        comp().setState({ startAnimations: false })
        expect(comp().find('.Greeter__headline_warning').hasClass('animate')).toBe(false)
      })
    })
  })

  describe('functions', () => {
    describe('handleLoad', () => {
      it('sets startAnimations to true', () => {
        comp().setState({ startAnimations: false })
        comp().instance().handleLoad()
        expect(comp().state('startAnimations')).toBe(true)
      })
    })

    describe('componentDidMount', () => {
      it('registers window load event listener', () => {
        const listeners = {}
        global.addEventListener = (name, fn) => listeners[name] = fn
        comp()
        expect(Object.keys(listeners)).toHaveLength(1)
      })

      it('sets documentElement className', () => {
        comp()
        expect(global.document.documentElement.className).toContain('html-dark-background')
      })
    })
  })

  it('matches snapshot', () => {
    expect(comp()).toMatchSnapshot()
  })
})

