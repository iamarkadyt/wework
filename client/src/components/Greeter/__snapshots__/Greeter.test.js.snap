// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Greeter matches snapshot 1`] = `
<div
  className="Greeter__container"
>
  <div
    className="Greeter__logo-screen false"
  >
    <div
      className="Greeter__logo"
    >
      <span>
        We
      </span>
      <span>
        Work
      </span>
    </div>
    <span
      className="Greeter__headline false"
    >
      Social network website for professionals
    </span>
    <span
      className="Greeter__headline_warning false"
    >
      Support for tall screens is
      <br />
      already on the way!
      <br />
      <br />
      <span
        style={
          Object {
            "fontSize": 80,
          }
        }
      >
        <FaShippingFast />
      </span>
      <br />
      <br />
      Until then, you can visit website
      <br />
      on desktop or switch your
      <br />
      device into landscape mode.
      <br />
      <br />
      To assess my skills in building
      <br />
      responsive layouts visit:
      <br />
      <a
        href="https://arkadyt.com"
        rel="noopener noreferrer"
        target="_blank"
      >
        <b>
          https://arkadyt.com
        </b>
      </a>
    </span>
  </div>
  <div
    className="Greeter__white-screen false"
  />
</div>
`;
